guides/source/active_record_encryption.md

```diff
@@ -51,71 +59,174 @@ config.active_record.encryption.key_derivation_salt = ENV["ACTIVE_RECORD_ENCRYPT
+## Basic Usage
 
-### Deterministic and Non-deterministic Encryption
+### Ignoring Case
 
-By default, Active Record Encryption uses a non-deterministic approach to encryption. Non-deterministic, in this context, means that encrypting the same content with the same password twice will result in different ciphertexts. This approach improves security by making crypto-analysis of ciphertexts harder, and querying the database impossible.
+You might need to ignore case when querying deterministically encrypted data. Two approaches make accomplishing this easier:
 
-You can use the `deterministic:`  option to generate initialization vectors in a deterministic way, effectively enabling querying encrypted data.
+You can use the `:downcase` option when declaring the encrypted attribute to downcase the content before encryption occurs.
 
 ```ruby
-class Author < ApplicationRecord
-  encrypts :email, deterministic: true
+class Person
+  encrypts :email_address, deterministic: true, downcase: true
+end
+```
+
+When using `:downcase`, the original case is lost. In some situations, you might
+want to ignore the case only when querying while also storing the original case.
+For those situations, you can use the `:ignore_case` option. This requires you
+to add a new column named `original_<column_name>` to store the content with the
+case unchanged:
+
+```ruby
+class Label
+  encrypts :name, deterministic: true, ignore_case: true # the content with the original case will be stored in the column `original_name`
 end
+```
+
+### Serialized Attributes
+
+By default, Active Record Encryption will serialize values using the underlying type before encrypting them as long as the value is serializable as Strings. If the underlying type is not serializable as a String, you can use a custom [`message_serializer`](https://edgeapi.rubyonrails.org/classes/ActiveRecord/Encryption/MessageSerializer.html):
+
+```ruby
+class Article < ApplicationRecord
+  encrypts :metadata, message_serializer: SomeCustomMessageSerializer.new
+end
+```
+
+Attributes with structured types using the [`serialized`](https://api.rubyonrails.org/v8.0.2/classes/ActiveRecord/AttributeMethods/Serialization/ClassMethods.html#method-i-serialize) method are supported with encryption as well. The `serialized` method is used when you have an attribute that needs to be saved to the database as a serialized object (using `YAML`, `JSON` or such), and retrieved by deserializing into the same object.
+
+NOTE: When using serialized attributes for custom types, the declaration of the serialized attribute should go **before** the encryption declaration:
 
-Author.find_by_email("some@email.com") # You can query the model normally
+```ruby
+# CORRECT
+class Article < ApplicationRecord
+  serialize :title, type: Title
+  encrypts :title
+end
+
+# INCORRECT
+class Article < ApplicationRecord
+  encrypts :title
+  serialize :title, type: Title
+end
+```
```

Comment:
Could this be simplified as:

```suggestion
```ruby
class Article < ApplicationRecord
  serialize :title, type: Title
  # serialized attributes are declared before encryption
  encrypts :title
end
```
```